tuf/src/repo_builder.rs

```diff
@@ -806,6 +819,18 @@ where
             })?;
 
             targets_builder = targets_builder.version(next_version);
+
+            // Insert all the metadata from the trusted snapshot.
+            if self.state.inherit_from_trusted_targets {
+                for (target_path, target_description) in trusted_targets.targets() {
+                    targets_builder = targets_builder
+                        .insert_target_description(target_path.clone(), target_description.clone());
+                }
+
+                if let Some(delegations) = trusted_targets.delegations() {
+                    targets_builder = targets_builder.delegations(delegations.clone());
```

Comment:
not sure if how much difference it makes but since is this a clone, we could avoid doing it here if `self.state.delegations` is some, since it will get overwritten anyway

Reply:
I ended up fixing this in a later patch.